src/plugins/slack/fetch.js

```diff
@@ -0,0 +1,168 @@
+// @flow 
+
+const { WebClient } = require('@slack/web-api')
+import type { 
+  SlackToken, 
+  User, 
+  Conversation, 
+  Message
+} from "./models"
+
+export class Fetcher {
+  +_token: SlackToken;
+  web: any
+
+  constructor(token: SlackToken) {
+    this._token = token
+    if (!token) {
+      throw new Error ("A Slack bot token is required");
+    }
+    this.web = new WebClient(token)
+  }
+
+  /**
+   * Get all users from the Slack organisation
+   */
+  async users(): Promise<Array<User>> {
+    const users: any = [];
+    let cursor: any;
+    let response: any;
+    do {
+      response = await this.web.users.list({
+        limit: 100,
+        include_locale: false,
+        cursor
+      });
+      const onlyHumans = await response.members.filter((member) => {
+        return member.is_bot !== true && member.profile.real_name_normalized !== 'Slackbot';
+      })
+      for (const human of onlyHumans) {
+        const oneHuman = {
+          id: human.id, 
+          name: human.profile.real_name_normalized,
+          email: human.profile.email 
+        };
+        users.push(oneHuman);
+      }
+      cursor = response.response_metadata.next_cursor;
+    } while (response.response_metadata.next_cursor);
+    return users;
+  }
+
+  /**
+   * Get all channels (public and private) from the Slack org
+   * Not considering Group DMs 
+   * @todo only public channels and private channels considered for now, include `type` type for public,private,group dms etc.
```

Comment:
Why not type this as `Array<Conversation>`?

Reply:
Just tried, it works that way!
my VSCode confused me with a red underline, I assumed flowjs wanted me to put in a promise there since the value comes from an async function response